f5_openstack_agent/lbaasv2/drivers/bigip/listener_service.py

```diff
@@ -203,13 +203,13 @@ def update_session_persistence(self, service, bigips):
             for bigip in bigips:
                 # For TCP listeners, must remove fastL4 profile before adding
                 # adding http/oneconnect profiles.
-                if listener['protocol'] == 'TCP':
-                    self._remove_profile(vip, 'fastL4', bigip)
+                if persistence_type != 'SOURCE_IP':
+                    if listener['protocol'] == 'TCP':
+                        self._remove_profile(vip, 'fastL4', bigip)
 
-                # Standard virtual servers should already have these profiles,
-                # but make sure profiles in place for all virtual server types.
-                self._add_profile(vip, 'http', bigip)
-                self._add_profile(vip, 'oneconnect', bigip)
+                    # HTTP listeners should have http and oneconnect profiles
+                    self._add_profile(vip, 'http', bigip)
+                    self._add_profile(vip, 'oneconnect', bigip)
```

Comment:
Right, and when the oneconnect profile is added, the TCP profile becomes /Common/tcp.